### ISSUE TITLE
Consider supporting iOS 7

### ISSUE THREAD

**Issue by irace** [tumblr/XExtensionItem#51]
It would just pass a single item through rather than providing any of the benefits, but at least would make it easier for apps that still support iOS 7 to simply drop XExtensionItem in.

**Comment by AriX** [comment]
I added basic iOS 7 compatibility in a547d68cc2ac5e4132bcbc18790f96570ffe0f97. There may well still be some other code lying around that could use iOS 8-only APIs, but this was all that was needed to get the NSExtensionItem demo working on iOS 7 for me.

**Comment by irace** [comment]
Great, thanks.

Do you think this is something we should do? I have mixed feelings. 

On one hand, it’d be easy enough and might help adoption.

On the other hand, I'm not even sure _how_ useful it is (in the Tumblr case, we have a code to create different Facebook/Twitter strings that include the URLs on iOS 7 where we can’t provide multiple attachments, so we probably wouldn't even use this ourselves). Not to mention I don’t really want to encourage supporting iOS 7.

**Comment by mattbischoff** [comment]
The more I think of it, the more I think we should do it. Not necessarily promote it anywhere, but just in order to make it easier to include this pod as a dependency of iOS 7+ projects.

**Comment by irace** [comment]
This is more work than I let on since it’d require redoing the Xcode project to work as a static library and not a framework :confused: 

Leaning towards “not for 1.0” as a result.

**Comment by AriX** [comment]
A few thoughts:
* I think it's a really good idea to support iOS 7. It's a responsible thing to do. Otherwise we could discourage adoption among developers who support iOS 7, which is a bunch of people especially since iOS 7 still has a pretty large userbase.
* People can easily just use git submodules and include the .h and .m files in their project rather than using CocoaPods or Xcode framework dependencies. I much prefer including code directly for small libraries like this one, and I don't think this should necessarily get in the way of testing the code itself to ensure iOS 7 compatibility.
* That said, I think having a static library target is a good idea anyway. I haven't looked at the project config but I assume you could just switch the current framework target to be a static library (and optionally, add a new dynamic framework target that just links against the static library).

**Comment by irace** [comment]
I agree with all of this just don’t have any time to work on it before heading off on vacation. I'll let it up to @mattbischoff to decide if we either hold 1.0.0 until one of us has time to rearrange things, or if we prioritize this for a subsequent point release.
